android/java/org/chromium/chrome/browser/app/domain/KeyringModel.java

```diff
@@ -264,38 +287,63 @@ public void getAccounts(Callbacks.Callback1<AccountInfo[]> callback1) {
 
     public void addAccount(String accountName, @CoinType.EnumType int coinType,
             Callbacks.Callback1<Boolean> callback) {
-        final AccountInfo[] finalAccountInfos =
+        final AccountInfo[] accountInfoArray =
                 WalletUtils.getAccountInfosFromKeyrings(_mKeyringInfosLiveData.getValue())
                         .toArray(new AccountInfo[0]);
         mKeyringService.addAccount(accountName, coinType, result -> {
-            if (result) {
-                boolean hasExistingAccountType = false;
-                for (AccountInfo accountInfo : finalAccountInfos) {
-                    hasExistingAccountType = (accountInfo.coin == coinType);
-                    if (hasExistingAccountType) break;
-                }
-                if (!hasExistingAccountType) {
-                    mKeyringService.getKeyringInfo(
-                            getSelectedCoinKeyringId(coinType), updatedKeyringInfo -> {
-                                for (AccountInfo accountInfo : updatedKeyringInfo.accountInfos) {
-                                    if (accountInfo.coin == coinType) {
-                                        setSelectedAccount(accountInfo.address, coinType);
-                                        break;
-                                    }
-                                }
-                            });
-                }
-            }
-            mCryptoSharedActions.updateCoinType();
-            mCryptoSharedActions.onNewAccountAdded();
-            callback.call(result);
+            handleAddAccountResult(result, accountInfoArray, coinType, callback);
+        });
+    }
+
+    public void addFilecoinAccount(String accountName, @FilecoinNetworkType String filecoinNetwork,
+            Callbacks.Callback1<Boolean> callback) {
+        final AccountInfo[] accountInfoArray =
+                WalletUtils.getAccountInfosFromKeyrings(_mKeyringInfosLiveData.getValue())
+                        .toArray(new AccountInfo[0]);
+        mSelectedFilecoinNetwork = filecoinNetwork;
+        mKeyringService.addFilecoinAccount(accountName, mSelectedFilecoinNetwork, result -> {
+            handleAddAccountResult(result, accountInfoArray, CoinType.FIL, callback);
```

Comment:
**Update:** discussed a solution on Slack, and I'm going to remove `accountInfoArray` and simplify the `handleAddAccountResult` method.

TL;DR
The new account is already selected after its creation in `KeyringService#AddAccount/AddFilecoinAccount`. Looping through all the account info array is not required. What **must** be called though are the methods

* `mCryptoSharedActions.updateCoinType()`
* `mCryptoSharedActions.onNewAccountAdded()`

This will update our data layer and notify live data attached.
Desktop is following a similar approach.